lib/rules/no-unused-state.js

```diff
@@ -101,7 +101,7 @@ module.exports = {
           parent &&
           parent.type === 'MethodDefinition' && (
             parent.static && parent.key.name === 'getDerivedStateFromProps' ||
-            classMethods.indexOf(parent.key.name !== -1)
+            classMethods.indexOf(parent.key.name) !== -1
```

Comment:
this seems like a legit bug, can we fix this with a regression test in another PR, first?

Reply:
Still hoping to see this peeled out, with a test, into a separate PR.

Reply:
PR #2274 .